Game/LambdaHack/Server/Commandline.hs

```diff
@@ -9,162 +12,277 @@ import Game.LambdaHack.Common.Prelude
-        in debugSer {sdebugCli = (sdebugCli debugSer) {sdisableAutoYes = True}}
-      parseArgs ("--noAnim" : rest) =
-        let debugSer = parseArgs rest
-        in debugSer {sdebugCli = (sdebugCli debugSer) {snoAnim = Just True}}
-      parseArgs ("--savePrefix" : rest) =
-        let (params, args) = breakPar rest
-            debugSer = parseArgs args
-        in debugSer { ssavePrefixSer = params
-                    , sdebugCli =
-                        (sdebugCli debugSer) {ssavePrefixCli = params}}
-      parseArgs ("--frontendTeletype" : rest) =
-        let debugSer = parseArgs rest
-        in debugSer {sdebugCli = (sdebugCli debugSer)
-                                    {sfrontendTeletype = True}}
-      parseArgs ("--frontendNull" : rest) =
-        let debugSer = parseArgs rest
-        in debugSer {sdebugCli = (sdebugCli debugSer) {sfrontendNull = True}}
-      parseArgs ("--frontendLazy" : rest) =
-        let debugSer = parseArgs rest
-        in debugSer {sdebugCli = (sdebugCli debugSer) {sfrontendLazy = True}}
-      parseArgs ("--dbgMsgCli" : rest) =
-        let debugSer = parseArgs rest
-        in debugSer {sdebugCli = (sdebugCli debugSer) {sdbgMsgCli = True}}
-      parseArgs (wrong : _rest) =
-        error $ "Unrecognized: " ++ wrong ++ "\n" ++ unlines usage
-  in parseArgs
+-- | Parser for server debug parameters from commandline arguments.
+debugModeSerPI :: ParserInfo DebugModeSer
+debugModeSerPI =  info ( helper <*> debugModeSerP )
+          $  fullDesc
+          <> progDesc "Configure debug options here, gameplay options in config.rules.ini."
+
+debugModeSerP :: Parser DebugModeSer
+debugModeSerP = do
+  knowMap              <- knowMapP
+  knowEvents           <- knowEventsP
+  knowItems            <- knowItemsP
+  sniffIn              <- sniffInP
+  sniffOut             <- sniffOutP
+  allClear             <- allClearP
+  boostRandomItem      <- boostRandItemP
+  gameMode             <- gameModeP
+  automateAll          <- automateAllP
+  keepAutomated        <- keepAutomatedP
+  ~(newGame,challenge) <- serToChallenge <$> newGameP
+  stopAfterSeconds     <- stopAfterSecsP
+  stopAfterFrames      <- stopAfterFramesP
+  benchmark            <- benchmarkP
+  dungeonRng           <- setDungeonRngP
+  mainRng              <- setMainRngP
+  dumpInitRngs         <- dumpInitRngsP
+  dbgMsgSer            <- dbgMsgSerP
+  gtkFontFamily        <- gtkFontFamilyP
+  sdlFontFile          <- sdlFontFileP
+  sdlTtfSizeAdd        <- sdlTtfSizeAddP
+  sdlFonSizeAdd        <- sdlFonSizeAddP
+  fontSize             <- fontSizeP
+  noColorIsBold        <- noColorIsBoldP
+  maxFps               <- maxFpsP
+  disableAutoYes       <- disableAutoYesP
+  noAnim               <- noAnimP
+  savePrefix           <- savePrefixP
+  frontendTeletype     <- frontendTeletypeP
+  frontendNull         <- frontendNullP
+  frontendLazy         <- frontendLazyP
+  dbgMsgCli            <- dbgMsgCliP
+
+  pure DebugModeSer
```

Comment:
Yes, you're right, wildCards will make it much easier. 
But there is an issue not to use  `defDebugModeCli`. Not all fields of `DebugModeCli` are covered by arguments: `stitle` and `sfontDir` are not.. Do we need to do anything about that?

Reply:
Oh, well spotted. Actually, it would be ideal to add a commandline option for `sfontDir`. And `stitle` is just a hack, so let's leave it empty: `let stitle = Nothing in DebugModeCli {..}`.